packages/styles/src/mixins/_utilities.scss

```diff
@@ -88,7 +88,8 @@
 }
 
 @mixin focus-style($border-radius: commons.$border-radius, $vendor-prefix: '') {
-  &:is(:focus-visible, :focus-within, .pretend-focus)#{$vendor-prefix} {
+  // :has(:focus-visible) mimic a focus-visible-within pseudo-class
+  &:is(:focus-visible, :has(:focus-visible), .pretend-focus)#{$vendor-prefix} {
     outline-offset: spacing.$size-line;
```

Comment:
If you want to get rid of that duplication as much as possible, you could use the long form definitions:

```scss
@mixin focus {
  outline-style: none;
  outline-offset: 2px;
  outline-width: 2px;
  outline-color: blue;
  
  &:where(:has(:focus-visible), :focus-visible) { outline-style: solid; @content; }
  @supports not selector(:has) { &:where(:focus-within) { outline-style: solid; @content; } }
  @include high-contrast-mode() {
     outline-color: Highlight;
  }
```

Reply:
Good idea, I did that except for hcm where specificity is easier to handle like that. Also `@supports not selector(:has)` is not working.